src/components/Error/ErrorBoundary.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import * as S from './ErrorBoundary.styles';
+
+type ErrorBoundaryProps = {
+  children: React.ReactNode;
+};
+
+type ErrorBoundaryState = {
+  hasError: boolean;
+  error: Error | null;
+};
+
+export class ErrorBoundary extends React.Component<
+  ErrorBoundaryProps,
+  ErrorBoundaryState
+> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false, error: null };
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    return { hasError: true, error };
+  }
+
+  componentDidCatch(error: Error, info: React.ErrorInfo) {
+    console.error(error, info);
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return <S.ErrorText>{this.state.error?.message}</S.ErrorText>;
```

Comment:
에러 바운더리는 사용하는 컴포넌트에서 fallback을 필요에 따라 지정할 수 있도록 설정하는게 좋습니다.
때문에 컴포넌트(ReactNode)를 props로 전달받고, 이를 fallback 컴포넌트로 렌더링하는게 좋을 것 같네요~
또한, 에러 메시지는 사용자 친화적이지 않은 내용일 가능성이 높기 때문에, 에러 메시지를 그대로 노출하는것도 권장되지 않습니다.
아래처럼 수정하는게 어떨까요?
```suggestion
      return this.props.fallback ?? <p>문제가 발생했습니다. 잠시 후 다시 시도해주세요.</p>;
```

Reply:
ErrorBoundary가 fallback 컴포넌트로 렌더링되도록 수정 완료했습니다.